### ISSUE TITLE
[Data Table keyboard nav]: if rows are not selectable and have overflow menu, keyboard navigation is broken

### ISSUE THREAD

**Issue by ZrianinaMariia** [carbon-design-system/carbon#4334]
## What package(s) are you using?

- [ ] `carbon-components`
- [x] `carbon-components-react` 7.6.4

## Detailed description

> Describe in detail the issue you're having.

If overflow menu is used inside a data table and rows are not selectable (i.e. do not have checkboxes in the first column), that keyboard navigation does not behave  in the expected way, i.e. when overflow menu is open and you press 'tab', the overflow menu items are still shown and it is not possible to close them with a keyboard, unless you press tab several time until the corresponding menu is on focus again and then press 'ESC' key.

> Is this issue related to a specific component?

OverflowMenu (and OVerflowMenuItem) inside a Data table

> What did you expect to happen? What happened instead? What would you like to
> see changed?

I expect it to behave the same way when rows are selectable, i.e. when I press tab, the overflow menu items will be hidden again and the overflow menu dots will get the focus.

> What browser are you working in?

Google Chrome Version 77.0.3865.90 

> What version of the Carbon Design System are you using?

10

> What offering/product do you work on? Any pressing ship or release dates we
> should be aware of?

Hyper Protect Services

## Steps to reproduce the issue

1. Click on the second overflow menu dots (second row)
2. Press tab key
> Please create a reduced test case in CodeSandbox

https://codesandbox.io/s/codesandbox-3551n

## Additional information

![image](https://user-images.githubusercontent.com/5481483/66831754-b12a2300-ef58-11e9-88d5-f7a45731cdfb.png)

**Comment by dakahn** [comment]
Hey, i was having problem reproducing here
http://react.carbondesignsystem.com/?path=/story/datatable--with-overflow-menu

Can I confirm that with you and if so we can figure out what you're missing from our latest updates maybe? 👍

**Comment by ZrianinaMariia** [comment]
@dakahn Thanks for looking into that. I probably was not clear enough in my description, sorry. The problem occurs when such a table does not have first column with checkboxes.

**Comment by ZrianinaMariia** [comment]
That's why I created a codesandbox to demonstrate the issue I am having.

https://codesandbox.io/s/codesandbox-3551n

**Comment by dakahn** [comment]
Can you reproduce using Carbon 7.7.3 (latest)?

**Comment by ZrianinaMariia** [comment]
When I switched to Carbon 7.7.3 in the codesandbox above, the tab navigation stop working at all, i.e. I could focus on first OverflowMenu and afterwards `tab` did nothing.

**Comment by emyarod** [comment]
I believe this is fixed in https://github.com/carbon-design-system/carbon/pull/4688

**Comment by asudoh** [comment]
Thanks @emyarod - Closing as I can verify what he says.
